docs/debugging-s2i.md

```diff
@@ -0,0 +1,116 @@
+
+	      W0720 21:07:52.145257 04204 sti.go:131 Clean build will be performed because of error saving previous build artifacts
+	      E0720 21:07:52.145263 04204 sti.go:133] ERROR: timeout waiting for tar stream
+
+
+Review the example [here](https://github.com/gabemontero/source-to-image/blob/master/docs/builder_image.md#save-artifacts), and perhaps revisit the tar man pages or the bash user manual, to address the problem.
+
+(GGM NOTE:  per Maciej's suggestion of reviewing errors.go, I pulled the NewTarTimeoutError as something worth highlighting, I also saw it referenced in
+https://github.com/openshift/source-to-image/issues/82... and I tried to incorporate some
+of the clarifications documented in the pull request.  )
+
+
+####Dowloading / Finding The Scripts
+
+Per this discussion point  [here](https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md#sti-scripts), if your s2i related scripts are already placed in the image, but their locations are not indicated by one of the provided means, attempts to download the scripts will fail.
+
+   	E0720 21:08:37.166063 04204 main.go:328] An error ocurred: scripts inside the image:  image:///path/to/image
+
+Properly reference the script location with the methods described in the above link.
+
+(GGM NOTE:  again. looking at errors.go, the NewScriptsInsideImageError caught my eye as a tricky one that could trip up users.
+
+I also went through the issue which I think helped drive this error code:  https://github.com/openshift/source-to-image/issues/75
+As well as the verbiage around this in the doc:  https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md#sti-scripts
+And lastly, followed to code flow from sti.go:Prepare(),  to install.go:InstallRequired, to install.go:run(), to download.go:Download() and download.go:Read().
+
+I'll admit that while I *think* I have nuances down, I'm not 100% sure, as I've iterated over these 3 pieces a few times ... for example, in the second link, the relevance
+of " then the --destination flag or the io.openshift.s2i.destination label applies only to sources and artifacts" is.  I *think* that
+means that sti only downloads the source repos, images, and whatever the scripts might download, but not the scripts themselves, as they are of course
+already included locally and don't need to be downloaded)
+
+
+####ONBUILD
+
+At various points when executing ONBUILD instructions, if those instructions result in the need for root user access, and the s2i builder has not been invoked to allow
+it to run as root, you will get the expected permissions errors for such scenarios.
+
+(GGM NOTE:  from the errors.go NewBuilderRootNotAllowedError path)
+
+Must Gather
```

Comment:
OK, it's here, so my previous comment from the top of this file about must haves does not applies anymore.